nomenclature/core.py

```diff
@@ -8,11 +10,49 @@
 
 @validate_arguments(config=dict(arbitrary_types_allowed=True))
 def process(
-    df: IamDataFrame,
+    df: pyam.IamDataFrame,
     dsd: DataStructureDefinition,
-    dimensions: list = None,
-    processor: RegionProcessor = None,
-):
+    dimensions: Optional[List[str]] = None,
+    processor: Optional[RegionProcessor] = None,
+) -> pyam.IamDataFrame:
+    """Function for validation and region aggregation in one step
+
+    Since there are some subtleties in the order of applying region processing and
+    validation, this function is the recommended way of using the nomenclature package
+    as it takes care of subtleties.
+
+    Parameters
+    ----------
+    df : pyam.IamDataFrame
+        Input data to be validated and aggregated.
+    dsd : DataStructureDefinition
+        Data templates that are used for validation.
```

Comment:
The term "data template" is used on the index page but nowhere else - we should aim to use consistent terms to describe the classes and use cases

Reply:
What's a better term then? Should I just change it to 'Used for validation' (leave out the "data templates")?

Reply:
The docstring of the class uses "Definition of datastructure codelists". Maybe use
> Definitions of codelists (usually project-specific) used for validation

Reply:
Perfect, thanks, I'll use this. I'll also update it in the index page for consistency.